src/seq.rs

```diff
@@ -91,6 +94,52 @@ pub trait SliceRandom {
     fn choose_multiple<R>(&self, rng: &mut R, amount: usize) -> SliceChooseIter<Self, Self::Item>
         where R: Rng + ?Sized;
 
+    /// Similar to [`choose`], but each item in the slice don't have the same
+    /// likelyhood of getting returned. The likelyhood of a given item getting
+    /// returned is proportional to the value returned by the mapping function
```

Comment:
I think it's spelled [likelihood](https://en.wiktionary.org/wiki/likelihood). Also, "don't have" is bad grammar and the sentence is indirectly describing the purpose. Try:

> A variant of `choose` where the likelihood of each outcome may be specified. The specified function `func` maps items `x` to a relative likelihood `func(x)`. The probability of each item being selected is therefore `func(x) / S`, where `S` is the sum of all `func(x)`.

Perhaps also rename `func` → `weight` or `w` or `f`?